recipes/api/tests/test_recipe_api.py

```diff
@@ -0,0 +1,214 @@
+from django.test import TestCase
+from django.urls import reverse
+from rest_framework.test import APIClient
+from rest_framework import status
+
+from api.serializers import RecipeSerializer
+from core.models import Recipe, Ingredient
+
+RECIPES_URL = reverse("api:recipes-list")
+
+
+def detail_url(recipe_id):
+    """Return recipe detail URL
+
+    :param recipe_id:
+    :return:
+    """
+    return reverse("api:recipe-detail", args=[recipe_id])
+
+
+def sample_recipe(name, **params):
+    """Helper function to create recipes
+
+    :param name:
+    :return:
+    """
+    ingredients = params.pop("ingredients", None)
+    recipe = Recipe.objects.create(name=name, **params)
+    if ingredients:
+        for name in ingredients:
+            recipe.ingredients.add(Ingredient.objects.create(name=name))
+
+    return recipe
+
+
+class RecipeAPITests(TestCase):
+    """None
+
+    """
+
+    def setUp(self) -> None:
+        self.client = APIClient()
+
+    def test_retrieve_basic_recipe_list(self):
+        """Test retrieving recipes
+
+        :return:
+        """
+        sample_recipe(name="Bolognaise")
+
+        res = self.client.get(RECIPES_URL)
+
+        recipes = Recipe.objects.all().order_by("-name")
+        serializer = RecipeSerializer(recipes, many=True)
+
+        self.assertEqual(status.HTTP_200_OK, res.status_code)
+        self.assertEqual(serializer.data, res.data)
```

Comment:
IMO for this and the rest it is helpful to assert against a dict literal to ensure that the content of data of both is correct. The view just passes back the content created by the serializer, so comparing view to serializer output doesn't test that much.

Reply:
Got it, makes sense!